test/e2e/trigger_dependency_annotation_test.go

```diff
@@ -91,7 +91,7 @@ func TestTriggerDependencyAnnotation(t *testing.T) {
 	if brokerClass == eventing.MTChannelBrokerClassValue {
 		pingSource.Spec.SourceSpec.Sink.URI = &apis.URL{
 			Scheme: "http",
-			Host:   fmt.Sprintf("broker-ingress.%s.svc.cluster.local", resources.SystemNamespace),
+			Host:   fmt.Sprintf("broker-ingress.%s.svc", resources.SystemNamespace),
```

Comment:
I think we should stop setting the URI directly and set the pingSource.Spec.SourceSpec.Sink.Ref to the Broker since we have all the bits here already, and would make this more resilient if we wanted to test against other brokers. Then we can drop the if clause and just do it in L87.